src/cegis/cegis-util/program_helper.cpp

```diff
@@ -350,18 +350,23 @@ goto_programt::targett insert_preserving_source_location(
 goto_programt::targett insert_after_preserving_source_location(
     goto_programt &body, goto_programt::targett pos)
 {
-  const auto op=std::bind1st(std::mem_fun(&goto_programt::insert_after), &body);
-  return insert_preserving_source_location(pos, op);
+  return insert_preserving_source_location(
+    pos,
+    [&](goto_programt::const_targett target)
+    {
+      return body.insert_after(target);
+    });
 }
 
 goto_programt::targett insert_before_preserving_source_location(
     goto_programt &body, goto_programt::targett pos)
 {
-  typedef goto_programt::targett (goto_programt::*ftype)(
-      goto_programt::targett);
-  const auto op=std::bind1st(
-      std::mem_fun(static_cast<ftype>(&goto_programt::insert_before)), &body);
-  return insert_preserving_source_location(pos, op);
+  return insert_preserving_source_location(
```

Comment:
This was broken due to `insert_before` changing to take a `const_targett` instead of a `targett`. I could just fix the typedef, but Meyers recommends in 'Effective Modern C++' that we prefer lambdas over `std::bind`, so I've made that change too.